.github/report-flaky-tests/index.js

```diff
@@ -0,0 +1,249 @@
+		);
+		const testResults = flakyTests[ flakyTestKey ];
+		const isTrunk = getHeadBranch() === 'trunk';
+
+		if ( reportedIssue ) {
+			let body = reportedIssue.body;
+			const meta = metaData.get( body );
+
+			if ( isTrunk ) {
+				const headCommit = github.context.sha;
+				const baseCommit = meta.baseCommit || github.context.sha;
+
+				let totalCommits = 0;
+				try {
+					const { data } = await octokit.rest.repos.compareCommits( {
+						...github.context.repo,
+						base: baseCommit,
+						head: headCommit,
+						per_page: 1,
+					} );
+
+					meta.failedTimes += testResults.length;
+					meta.totalCommits = data.total_commits + 1;
+				} catch ( err ) {
+					// It might a forced push or a deleted commit,
+					// we have to treat the current commit as the base commit.
+					meta.baseCommit = headCommit;
+					meta.failedTimes = testResults.length;
+					meta.totalCommits = 1;
+				}
+			}
+
+			body =
+				renderIssueDescription( { meta, testTitle, testPath } ) +
+				body.slice(
+					body.indexOf( TEST_RESULTS_LIST.open ),
+					body.indexOf( TEST_RESULTS_LIST.close )
+				) +
+				[
+					renderTestErrorMessage( { testPath, testResults } ),
+					TEST_RESULTS_LIST.close,
+				].join( '\n' );
+
+			await octokit.rest.issues.update( {
+				...github.context.repo,
+				issue_number: reportedIssue.number,
+				body,
+			} );
+		} else {
+			const meta = isTrunk
+				? {
+						failedTimes: testResults.length,
+						totalCommits: 1,
+						baseCommit: github.context.sha,
+				  }
+				: {
+						failedTimes: 0,
+						totalCommits: 0,
+				  };
+
+			await octokit.rest.issues.create( {
+				...github.context.repo,
+				title: issueTitle,
+				body: renderIssueBody( {
+					meta,
+					testTitle,
+					testPath,
+					testResults,
+				} ),
+				labels: [ label ],
+				// TODO: Maybe we can combine git blame to automatically assign issue to the original author?
+			} );
+		}
+	}
+} )().catch( ( err ) => {
+	core.error( err );
+} );
+
+async function downloadReportFromArtifact( octokit, artifactName ) {
+	const {
+		data: { artifacts },
+	} = await octokit.rest.actions.listWorkflowRunArtifacts( {
+		...github.context.repo,
+		run_id: github.context.payload.workflow_run.id,
+	} );
+
+	const matchArtifact = artifacts.find(
+		( artifact ) => artifact.name == artifactName
+	);
+
+	if ( ! matchArtifact ) {
+		return;
+	}
+
+	const download = await octokit.rest.actions.downloadArtifact( {
+		...github.context.repo,
+		artifact_id: matchArtifact.id,
+		archive_format: 'zip',
+	} );
+
+	const {
+		files: [ file ],
+	} = await unzipper.Open.buffer( Buffer.from( download.data ) );
+	const buffer = await file.buffer();
+
+	return JSON.parse( buffer.toString() );
+}
+
+function getIssueTitle( testTitle ) {
+	return `[Flaky Test] ${ testTitle }`;
+}
+
+function getTestPathAndTitle( testKey ) {
+	const separatorIndex = testKey.indexOf( ':' );
+	return [
+		testKey.slice( 0, separatorIndex ),
+		testKey.slice( separatorIndex + 1 ),
+	];
```

Comment:
Could this be `return testKey.split( ':' )?`

Reply:
That won't work if the test has `:` in its title.